### PR TITLE
feat: introduce cached debug for better errors

### PR DESCRIPTION
**Pull request by timsuchanek**

### CODE REVIEW

File: src/packages/.eslintrc.js [hunk 1]
```diff
@@ -35,6 +35,7 @@ module.exports = {
                 '@typescript-eslint/no-use-before-define': 'off',
                 '@typescript-eslint/no-non-null-assertion': 'off',
                 'no-useless-escape': 'off',
+                '@typescript-eslint/no-explicit-any': 'off'
```
Comment (Jolg42) [inline src/packages/.eslintrc.js:38]:
```suggestion
```
Can you revert this maybe and use this:

```
line // eslint-disable-line @typescript-eslint/no-explicit-any
or
// eslint-disable-next-line @typescript-eslint/no-explicit-any
line
```

File: src/packages/sdk/package.json [hunk 2]
```diff
@@ -39,7 +39,7 @@
     "prettier": "^2.0.4",
     "ts-jest": "^25.3.1",
     "ts-node": "^8.8.2",
-    "typescript": "^3.8.3"
+    "typescript": "^4.0.0-dev.20200512"
```
Comment (Jolg42) [inline src/packages/sdk/package.json:42]: They released 3.9.2 just 13 hours ago :P Not sure why you need the 4 dev?

File: src/packages/sdk/src/IntrospectionEngine.ts [hunk 3]
```diff
@@ -183,16 +183,11 @@ export class IntrospectionEngine {
...
           const binaryPath = await resolveBinary('introspection-engine')
           debugRpc('starting introspection engine with binary: ' + binaryPath)
           this.child = spawn(binaryPath, {
             stdio: ['pipe', 'pipe', 'pipe'],
-            env: {
-              ...env,
-              // RUST_LOG: 'info',
-              // RUST_BACKTRACE: '1',
-            },
+            env: process.env,
```
Comment (Jolg42) [inline src/packages/sdk/src/IntrospectionEngine.ts:190]: Looks unrelated to debug, does it fix something or it's a cleanup?

Comment (timsuchanek) [reply src/packages/sdk/src/IntrospectionEngine.ts:190]: This is cleanup

File: src/packages/sdk/package.json (hunk 2 above)
Comment (timsuchanek) [reply src/packages/sdk/package.json:42]: Yep, tried the dev one out because of a bug, but I can go back to 3.8.3

File: src/packages/.eslintrc.js (hunk 1 above)
Comment (timsuchanek) [reply src/packages/.eslintrc.js:38]: Ok :)

**Review by Jolg42 — APPROVED** [review]

File: src/packages/sdk/package.json (hunk 2 above)
Comment (timsuchanek) [reply src/packages/sdk/package.json:42]: It was btw this bad boy that I was debugging https://github.com/microsoft/TypeScript/issues/38532

File: src/packages/client/package.json [hunk 4]
```diff
@@ -50,6 +50,7 @@
   ],
   "devDependencies": {
     "@apexearth/copy": "^1.4.5",
+    "@prisma/debug": "workspace:^2.0.0-alpha.1205",
```
Comment (Jolg42) [inline src/packages/client/package.json:53]:
```suggestion
    "@prisma/debug": "workspace:*",
```

File: src/packages/cli/package.json [hunk 5]
```diff
@@ -49,12 +49,13 @@
...
   },
   "prisma": {
-    "version": "0c2ec197653b278b2978845ef958db88824cd82e"
+    "version": "947599a1413bc514e275905d562b50200bac990b"
   },
   "devDependencies": {
     "@apexearth/copy": "^1.4.5",
     "@prisma/ci-info": "^2.1.1",
     "@prisma/client": "workspace:*",
+    "@prisma/debug": "workspace:^2.0.0-alpha.1205",
```
Comment (Jolg42) [inline src/packages/cli/package.json:58]:
```suggestion
    "@prisma/debug": "workspace:*",
```

File: src/packages/debug/src/util.ts [hunk 6]
```diff
@@ -0,0 +1,6 @@
+export function removeISODate(str: string): string {
+  return str.replace(
+    /\d{4}-[01]\d-[0-3]\dT[0-2]\d:[0-5]\d:[0-5]\d\.\d+([+-][0-2]\d:[0-5]\d|Z)/gim,
```
Comment (Jolg42) [inline src/packages/debug/src/util.ts:3]:
```suggestion
    /\d{4}-[01]\d-[0-3]\dT[0-2]\d:[0-5]\d:[0-5]\d\.\d+([+-][0-2]\d:[0-5]\d|Z)/i,
```

If the ISO date is on the first line and at the beginning you probably want to remove global and multiline?

File: src/packages/engine-core/package.json [hunk 7]
```diff
@@ -6,6 +6,7 @@
   "types": "dist/index.d.ts",
   "author": "Tim Suchanek <suchanek@prisma.io>",
   "devDependencies": {
+    "@prisma/debug": "workspace:^2.0.0-alpha.1205",
```
Comment (Jolg42) [inline src/packages/engine-core/package.json:9]:
```suggestion
    "@prisma/debug": "workspace:*",
```

File: src/packages/fetch-engine/package.json [hunk 8]
```diff
@@ -6,6 +6,7 @@
   "license": "Apache-2.0",
   "author": "Tim Suchanek <suchanek@prisma.io>",
   "devDependencies": {
+    "@prisma/debug": "workspace:^2.0.0-alpha.1205",
```
Comment (Jolg42) [inline src/packages/fetch-engine/package.json:9]:
```suggestion
    "@prisma/debug": "workspace:*",
```

File: src/packages/debug/src/util.ts (hunk 6 above)
Comment (timsuchanek) [reply src/packages/debug/src/util.ts:3]: Nope, it's not just in the first line, as we're running it on a multiline stringified json.

File: src/packages/generator-helper/package.json [hunk 9]
```diff
@@ -36,6 +35,7 @@
     ]
   },
   "devDependencies": {
+    "@prisma/debug": "workspace:^2.0.0-alpha.1205",
```
Comment (Jolg42) [inline src/packages/generator-helper/package.json:38]:
```suggestion
    "@prisma/debug": "workspace:*",
```

File: src/packages/get-platform/package.json [hunk 10]
```diff
@@ -6,6 +6,7 @@
   "license": "Apache-2.0",
   "author": "Tim Suchanek <suchanek@prisma.io>",
   "devDependencies": {
+    "@prisma/debug": "workspace:^2.0.0-alpha.1205",
```
Comment (Jolg42) [inline src/packages/get-platform/package.json:9]:
```suggestion
    "@prisma/debug": "workspace:*",
```

File: src/packages/debug/src/util.ts (hunk 6 above)
Comment (timsuchanek) [reply src/packages/debug/src/util.ts:3]: And `i` never hurts :)

File: src/packages/ink-components/package.json [hunk 11]
```diff
@@ -23,6 +22,7 @@
     "dist"
   ],
   "devDependencies": {
+    "@prisma/debug": "workspace:^2.0.0-alpha.1205",
```
Comment (Jolg42) [inline src/packages/ink-components/package.json:25]:
```suggestion
    "@prisma/debug": "workspace:*",
```

File: src/packages/migrate/package.json [hunk 12]
```diff
@@ -11,6 +11,7 @@
   },
   "devDependencies": {
     "@prisma/client": "workspace:*",
+    "@prisma/debug": "workspace:^2.0.0-alpha.1205",
```
Comment (Jolg42) [inline src/packages/migrate/package.json:14]:
```suggestion
    "@prisma/debug": "workspace:*",
```

File: src/packages/sdk/package.json [hunk 13]
```diff
@@ -20,9 +20,10 @@
     "scripts"
   ],
   "prisma": {
-    "version": "0c2ec197653b278b2978845ef958db88824cd82e"
+    "version": "947599a1413bc514e275905d562b50200bac990b"
   },
   "devDependencies": {
+    "@prisma/debug": "workspace:^2.0.0-alpha.1205",
```
Comment (Jolg42) [inline src/packages/sdk/package.json:26]:
```suggestion
    "@prisma/debug": "workspace:*",
```

File: src/pnpm-lock.yaml [hunk 14]
```diff
@@ -96,6 +97,7 @@ importers:
       '@apexearth/copy': ^1.4.5
       '@prisma/ci-info': ^2.1.1
       '@prisma/client': 'workspace:*'
+      '@prisma/debug': 'workspace:^2.0.0-alpha.1205'
```
Comment (Jolg42) [inline src/pnpm-lock.yaml:100]:
```suggestion
      '@prisma/debug': 'workspace:*'
```

File: src/pnpm-lock.yaml [hunk 15]
```diff
@@ -213,6 +216,7 @@ importers:
       webpack-cli: 4.0.0-alpha-4_webpack@5.0.0-alpha.14
     specifiers:
       '@apexearth/copy': ^1.4.5
+      '@prisma/debug': 'workspace:^2.0.0-alpha.1205'
```
Comment (Jolg42) [inline src/pnpm-lock.yaml:219]:
```suggestion
      '@prisma/debug': 'workspace:*'
```

File: src/packages/sdk/package.json (hunk 2 above)
Comment (Jolg42) [reply src/packages/sdk/package.json]: Ok Thanks! :)

File: src/packages/debug/src/util.ts (hunk 6 above)
Comment (Jolg42) [reply src/packages/debug/src/util.ts:3]: Ok 👍

**Review by Jolg42 — APPROVED** [review]
Nice 👍
